src/main/java/ch/uzh/ifi/hase/soprafs22/controller/InGameWebsocketController.java

```diff
@@ -15,18 +17,26 @@
 @Controller
 public class InGameWebsocketController {
 
+    private final InGameWebsocketService service;
+
+    InGameWebsocketController(InGameWebsocketService service) {
+        this.service = service;
+    }
+    
     @MessageMapping("/move")
     @SendTo("/client/move")
-    public ExampleMoveGetDTO exampleMove(ExampleMovePostDTO exampleMovePostDTO, Principal principal) throws Exception {
+    public MoveGetDTO exampleMove(MovePostDTO exampleMovePostDTO, Principal principal, Long gameId) throws Exception {
         // get move from the client
-        ExampleMove move = DTOMapper.INSTANCE.convertExampleMovePostDTOtoEntity(exampleMovePostDTO);
+        Move move = DTOMapper.INSTANCE.convertMovePostDTOtoEntity(exampleMovePostDTO);
 
         // verify move validity and add Player details
         String username = principal.getName();
-        move.setPlayer(new Player(username, 2, "randomUUID", "fake password", Color.BLUE));
+        move = service.verifyMove(move, username, gameId);
 
         // notify subscribers with the move
-        return DTOMapper.INSTANCE.convertEntityToExampleMoveGetDTO(move);
+        /* TODO: Should this return the whole gameState instead of only a move? 
```

Comment:
I think the game state should be sent over an url like:
`@SendTo("/client/state")` 
or something like 
`@SendTo("/client/GREEN/state")` 
in order to only notify a specific player (here green).

Reply:
for every move I would just notify all players with a move message.
(even the player moving will only execute the move upon receiving the move message)

Reply:
But then how do all the players get the updated gamestate after a move is made?

Reply:
They should update the state locally themselves. This is also how most games function with TCP/IP.